### PR TITLE
[FIX] account: payment wizard available journals with one payment type

### PR DESCRIPTION
**Pull request by agr-odoo**
Create an invoice to partner A, confirm
Create an invoice to partner B, confirm
From Invoice list view, select just one invoice and hit register payment
It will be possible to only select journals having inbound payment
method
Now select both invoices and hit register payment
It will be possible to select any Bank journal, even if both invoice
payment should be incoming

opw-3757686

### CODE REVIEW

**Comment by agr-odoo** [comment]
@smetl Just a tiny consistency fix, not sure it is necessary
In 15.0 we used to always show all Bank journals

File: addons/account/wizard/account_payment_register.py [hunk 1]
```diff
@@ -380,7 +380,8 @@ def _compute_currency_id(self):
     @api.depends('payment_type', 'company_id', 'can_edit_wizard')
     def _compute_available_journal_ids(self):
         for wizard in self:
-            if wizard.can_edit_wizard:
+            single_payment_type = len(set(batch['payment_values']['payment_type'] for batch in wizard._get_batches())) == 1
+            if wizard.can_edit_wizard or single_payment_type:
                 batch = wizard._get_batches()[0]
                 wizard.available_journal_ids = wizard._get_batch_available_journals(batch)
```
Comment (smetl) [inline addons/account/wizard/account_payment_register.py:386]: Technically, it's working but I really don't like calling that part with the first batch you find. Maybe it's more logical to call `_get_batch_available_journals` for each batch and aggregate all the results together instead

File: addons/account/wizard/account_payment_register.py [hunk 2]
```diff
@@ -380,14 +380,10 @@ def _compute_currency_id(self):
...
-                wizard.available_journal_ids = wizard._get_batch_available_journals(batch)
-            else:
-                wizard.available_journal_ids = self.env['account.journal'].search([
-                    ('company_id', '=', wizard.company_id.id),
-                    ('type', 'in', ('bank', 'cash')),
-                ])
+            available_journal_ids = self.env['account.journal']
+            for batch in wizard._get_batches():
+                available_journal_ids |= wizard._get_batch_available_journals(batch)
+            wizard.available_journal_ids = wizard._get_batch_available_journals(batch)
```
Comment (smetl) [inline addons/account/wizard/account_payment_register.py:386]:
```suggestion
            available_journals = self.env['account.journal']
            for batch in wizard._get_batches():
                available_journals |= wizard._get_batch_available_journals(batch)
            wizard.available_journal_ids = [Command.set(available_journals.ids)]
```

Comment (agr-odoo) [reply addons/account/wizard/account_payment_register.py:386]: oh my...it seems I forgot to change the last line :sweat_smile:

**Pull request closed without merging**